libnetwork/drivers/bridge/setup_ip_tables_linux_test.go

```diff
@@ -374,3 +377,68 @@ func TestOutgoingNATRules(t *testing.T) {
 		})
 	}
 }
+
+func TestMirroredWSL2Workaround(t *testing.T) {
+	// Avoid overwriting a real "/usr/bin/wslinfo" (and clashing with a real
+	// loopback0).
+	if _, err := os.Stat(wslinfoPath); err == nil {
+		t.Skip("Skipping test because " + wslinfoPath + " exists")
+	}
+
+	for _, tc := range []struct {
+		desc             string
+		loopback0        bool
+		wslinfoExists    bool
+		wslinfoPerm      os.FileMode // 0 for no-file
+		expLoopback0Rule bool
+	}{
+		{
+			desc: "No loopback0",
+		},
+		{
+			desc:             "WSL2 mirrored",
+			loopback0:        true,
+			wslinfoPerm:      0777,
+			expLoopback0Rule: true,
+		},
+		{
+			desc:        "loopback0 but wslinfo not executable",
+			loopback0:   true,
+			wslinfoPerm: 0666,
+		},
+		{
+			desc:      "loopback0 but no wslinfo",
+			loopback0: true,
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+			defer netnsutils.SetupTestOSContext(t)()
+
+			if tc.loopback0 {
+				loopback0 := &netlink.Dummy{
+					LinkAttrs: netlink.LinkAttrs{
+						Name: "loopback0",
+					},
+				}
+				err := netlink.LinkAdd(loopback0)
+				assert.NilError(t, err)
+			}
+
+			if tc.wslinfoPerm != 0 {
+				wslinfoPathOrig := wslinfoPath
+				defer func() {
+					wslinfoPath = wslinfoPathOrig
+				}()
+				tmpdir := t.TempDir()
+				wslinfoPath = filepath.Join(tmpdir, "wslinfo")
+				err := os.WriteFile(wslinfoPath, []byte("#!/bin/sh\necho dummy file\n"), tc.wslinfoPerm)
+				assert.NilError(t, err)
+			}
+
+			_, _, _, _, err := setupIPChains(configuration{EnableIPTables: true}, iptables.IPv4)
+			assert.NilError(t, err)
+			assert.Check(t, mirroredWSL2Rule().Exists() == tc.expLoopback0Rule,
+				fmt.Sprintf("Expected exists=%v", tc.expLoopback0Rule))
```

Comment:
Ah! I didn't spot it this time; good catch.

Using `is.Equal(mirroredWSL2Rule().Exists(), tc.expLoopback0Rule)` will also do all of that; it prints the variable names in failure, so if those are descriptive, it will be printing something like;

```
    foo_test.go:35: assertion failed: false (bool) != true (tc.expLoopback0Rule bool)
```

Reply:
Swapped for `is.Equal`.